Update log_custom_fields_by_lua.md with note of using dots (#8486)
Kong/docs.konghq.com: app/_includes/md/plugins-hub/log_custom_fields_by_lua.md

  --data config.custom_fields_by_lua.header="return kong.request.get_header('h1')"
```

### Special characters

> Dots `.` in the field key creates nested fields. You can use a backslash `\` to escape dots if you want to keep them in the field name.

Reviewer: ```suggestion
Dot characters (`.`) in the field key create nested fields. You can use a backslash `\` to escape a dot if you want to keep it in the field name.
```

> For example, if you configure a field in File Log with both dot and escaped dot:

Reviewer: ```suggestion
For example, if you configure a field in the File Log plugin with both a regular dot and an escaped dot:
```

```sh
curl -i -X POST http://localhost:8001/plugins/ \
...
  --data config.name=file-log \
  --data config.custom_fields_by_lua[my_file.log\.field]="return foo"
```

> You will see below field in file log:

Reviewer: ```suggestion
The field will look like this in the log:
```
